aten/src/ATen/native/cuda/Lerp.cu

```diff
@@ -33,11 +74,12 @@ void lerp_tensor_kernel(at::TensorIteratorBase& iter) {
                           (opmath_t{1} - weight_val_f);
             });
       });
+  }
 }
 
 void lerp_scalar_kernel(at::TensorIteratorBase& iter, const c10::Scalar& weight) {
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(
-      at::ScalarType::Half, at::ScalarType::BFloat16,
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND3(
+      kComplexHalf, at::ScalarType::Half, at::ScalarType::BFloat16,
```

Comment:
We should use jiterator for this kernel as well.

Also I think you will need to pass weight (which is a runtime argument) to the kernel. See below

https://github.com/pytorch/pytorch/blob/88fca3be5924dd089235c72e651f3709e18f76b8/aten/src/ATen/native/cuda/JitLoops.cuh#L44-L55

Reply:
Updated it using jiterator to `lerp_scalar` kernel too. Thanks, Kshitij!